specs/text-field/text-field.md

```diff
@@ -0,0 +1,142 @@
+### API
+
+Extends form associated custom element
+
+*Component Name*
+- `fast-text-field`
+
+*Attrs* - Adapted from [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/text)
+- `readonly` - the text field should be submitted with the form but should not be editable.
+- `placeholder` - an exemplar value to display in the input field whenever it is empty
+- `required` - boolean value that sets the field as required
+- `value` - string value of the text field, can be an empty string
+- `disabled` - disables the control
+- `list` - the id of the `<datalist>` element that contains the optional pre-defined autocomplete options
+- `maxlength`	- the maximum number of characters the input should accept
+- `minlength` -	the minimum number of characters long the input can be and still be considered valid
+- `pattern` - a regular expression the input's contents must match in order to be valid
+- `size` - a number indicating how many characters wide the input field should be
+- `spellcheck` - controls whether or not to enable spell checking for the input field, or if the default spell checking configuration should be used
+- `type` - enum
+  - text
+  - email
+  - password
+  - tel
+  - url
+- `appearance` - enum
+  - outline (default)
+  - filled
+
+*Events*
+- `change: CustomEvent`
+  - no custom data
+  - bubbles
+
+### Anatomy and Appearance
+
+```HTML
+<!-- shadow root -->
+<label part="label"><slot></slot></label>
+<div part="root">
+  <slot name="before-content" part="before-content"></slot>
+  <input type="${x => x.type}" part="control" />
+  <slot name="after-content" part="after-content"></slot>
+</div>
+<!-- end shadow root -->
+```
+
+
+*Screenshots below are of the basic appearance of the component and are not exhaustive.*
+
+| State | Image |
+| ----- | ----- |
+| default | ![](./images/text-field.png) |
+| focus | ![](./images/text-field-focus.png)
+| placeholder | ![](./images/text-field-placeholder.png)
+| password (filled) | ![](./images/text-field-password.png)
+| filled | ![](./images/text-field-filled.png)
+
+*Slot Names*
+- default - the label content
+- before-content - often times a glyph, icon, or button precedes input
+- after-content - often times a glyph, icon, or button follows the input
+
+*Host Classes*
+- disabled
+- required
+- readonly
+
+*CSS Parts*
+- label
+- root
+- before-content
+- after-content
+- control
+
+---
+
+## Implementation
+
+### States
+
+**disabled**: `true` or `false` - when disabled, the value will not be changeable through user interaction. It should also not expose it's value to a form submission.
+
+**readonly**: `true` or `false` - when readonly, the value will not be changeable through user interaction. The value will still be exposed to forms on submission.
+
+### Accessibility
+
+The input element inside the shadow-dom of the checkbox will be a focusable element with the following accessability content attributes:
+- role: [textbox](https://www.w3.org/WAI/PF/aria/roles#textbox)
```

Comment:
I don't think you'll these attributes if you're bringing focus to the input itself - AT will recognize that focus is on an input element and respond accordingly